### PR TITLE
For #27468 - Backplate homepage MessageCard

### LINKED ISSUE #27468: Wallpapers - Adapt "set as default browser" button with wallpaper card color treatment
Follow-up to:
- https://github.com/mozilla-mobile/fenix/issues/26520

Dependency:
- https://github.com/mozilla-mobile/fenix/issues/27578

**Current state**

**TODO**
- Color the card as the appropriate light/dark card color
- Color the button and button text a la `RecentSyncedTab` 
    - Button as `Layer 1`
    - Button text as `textActionSecondary` only in the light theme

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21831)

### PR DESCRIPTION
**Pull request by MozillaNoah**
| Light | Dark |
| -- | -- |
|  |  |

In essence, this section is styled the same as the recent synced tab (Jump back in)

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [ ] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #27468

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt [hunk 1]
```diff
@@ -47,11 +54,32 @@ class MessageCardViewHolder(
     @Composable
     override fun Content() {
         val message by remember { mutableStateOf(messageGlobal) }
+        val wallpaperState = components.appStore
+            .observeAsComposableState { state -> state.wallpaperState }.value ?: WallpaperState.default
+        val isWallpaperNotDefault = !Wallpaper.nameIsDefault(wallpaperState.currentWallpaper.name)
+        val defaultMessageCardColors = MessageCardColors.buildMessageCardColors()
+
+        var buttonColor = defaultMessageCardColors.buttonColor
+        var buttonTextColor = defaultMessageCardColors.buttonTextColor
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt:63]:
```suggestion
        var (buttonColor, buttonTextColor) = MessageCardColors.buildMessageCardColors()
```
Wondering if we could use destructuring declaration here to save us a few lines https://kotlinlang.org/docs/destructuring-declarations.html

**Review by gabrielluong — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt (hunk 1 above)
Comment (MatthewTighe) [reply app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt:63]: If we wanted to go fully immutable you may be able to do something like this, even:

val (buttonColor, buttonTextColor) = if (isWallpaperNotDefault) {
    FirefoxTheme.colors.layer1 to if (!isSystemInDarkTheme()) { FirefoxTheme.colors.textActionSecondary } else defaultMessageCardColors.buttonTextColor
} else {
   MessageCardColors.buildMessageCardColors()
}

**Review by MatthewTighe — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt (hunk 1 above)
Comment (MozillaNoah) [reply app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt:63]: I had no idea about the destructuring! that's really cool! I had to just insert some underscores though, as the destructuring is tied to the ordering of the parameters.

```
var (_, _, _, _, buttonColor, buttonTextColor) = MessageCardColors.buildMessageCardColors()
```

**Comment by Kate-Galetski** [comment]
Looks Correct!

**Pull request merged**